waffle-jest/test/matchers/events.test.ts

```diff
@@ -0,0 +1,66 @@
+import {Contract, ContractFactory} from 'ethers';
+import {MockProvider} from '@ethereum-waffle/provider';
+import {EVENTS_ABI, EVENTS_BYTECODE} from '../contracts/Events';
+
+describe('INTEGRATION: Events Emitted', () => {
+  const [wallet] = new MockProvider().getWallets();
+  let factory: ContractFactory;
+  let events: Contract;
+
+  beforeEach(async () => {
+    factory = new ContractFactory(EVENTS_ABI, EVENTS_BYTECODE, wallet);
+    events = await factory.deploy();
```

Comment:
Similarily, this pattern can also be changed to using a fixture, speeding up the tests.

Reply:
See above.